### DISCUSSION TITLE
Send/Receive message during call

### DISCUSSION THREAD

**Discussion by mgsray** [sipsorcery-org/sipsorcery#585] [Q&A]
Guys,
I have a requirement to receive/send message from/to sip server during sip call. The message is text base, can be treated as notification. 
Currently, I use sipuseragent, it works on the call part. What should I do to realize the message function?

**Comment by sipsorcery** [comment]
There is no feature that will let you send arbitrary in dialog SIP requests, which I think is what you're asking. The closest is the on/off hold [reINVITE requests](https://github.com/sipsorcery-org/sipsorcery/blob/cbe0bf7343aece9c902895356a0903e46ab9d4d3/src/app/SIPUserAgents/SIPUserAgent.cs#L1198). 

You could use that method as a starting point and change the request type to MESSAGE, or whatever you require, and it should work.
